Automattic/wp-calypso
# Jetpack Settings: Implement bulk settings management into Redux

### Introduction

This PR introduces actions, reducer and selectors for bulk managing (retrieving and updating) Jetpack module settings. Also, a query component for managing network requests for bulk module settings is being introduced. This PR is part of the Jetpack Module Settings group effort (#9171). 

This PR will allow us to update module settings and activate/deactivate modules in a single request; also to read all module settings in a single request. 

### New WP.COM API methods
For interacting with the .com API it introduces 2 methods that are not yet implemented on the API side:

* `fetchJetpackSettings()`
* `updateJetpackSettings()`

In order to fully test these methods you can:
* Apply D3588-code on your .com sandbox
* Checkout https://github.com/Automattic/jetpack/pull/5418 on your Jetpack site.
* Test with the latest Jetpack to make sure you have https://github.com/Automattic/jetpack/pull/6002 on your Jetpack site (it introduces the settings retrieval endpoint).
* Connect your Jetpack site to your .com sandbox and verify they talk to each other.

### To test

* Checkout this branch
* Play with the `QueryJetpackSettings` query component and verify it dispatches the proper network requests - `<QueryJetpackSettings siteId={ 1234 } />`.
* Verify all tests are working: `npm run test-client client/state/jetpack-settings/settings/`
